be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -586,7 +594,61 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
 Status HdfsOrcScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     _should_skip_file = false;
     _use_orc_sargs = true;
+    build_row_delete_set(scanner_params);
     // todo: build predicate hook and ranges hook.
     return Status::OK();
 }
+
+Status HdfsOrcScanner::build_row_delete_set(const HdfsScannerParams& scanner_params) {
+    for (const auto& tdelete_file : scanner_params.deletes) {
+        scan_pos_delete_file(*tdelete_file);
+    }
+
+    return Status::OK();
+}
+
+Status HdfsOrcScanner::scan_pos_delete_file(const TIcebergDeleteFile& delete_file) {
```

Comment:
Consider to put this method into `orc_chunk_reader` and let HdfsOrcScanner do simple things.